### PR TITLE
TRUNK-5343: Create an api-pre2.2 sub module for the coreapps module.

### PR DESCRIPTION
**Pull request by pluwum**
## Description of what I changed
This PR creates an api-pre2.2 sub module for the coreapps module.

## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-5343

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: omod/src/main/resources/config.xml [hunk 1]
```diff
@@ -58,6 +58,11 @@
 			<path>/lib/coreapps-api-2.2-${project.parent.version}.jar</path>
 			<openmrsVersion>2.2.*</openmrsVersion>
 		</conditionalResource>
+
+		<conditionalResource>
+			<path>/lib/coreapps-api-pre2.2-${project.parent.version}.jar</path>
+			<openmrsVersion>2.2.*</openmrsVersion>
```
Comment (dkayiwa) [inline omod/src/main/resources/config.xml]: This should be for versions from 1.9.* to 2.1.*

Comment (pluwum) [reply omod/src/main/resources/config.xml]: @dkayiwa i have made changes to the versions.

File: api-pre2.2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		<!-- Begin OpenMRS core -->
+
+		<dependency>
+			<groupId>org.openmrs.api</groupId>
+			<artifactId>openmrs-api</artifactId>
+			<type>jar</type>
+		</dependency>
+
+		<dependency>
+			<groupId>org.openmrs.web</groupId>
```
Comment (dkayiwa) [inline api-pre2.2/pom.xml]: web in the api project?

File: api-pre2.2/pom.xml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		</dependency>
+
+		<dependency>
+			<groupId>org.openmrs.api</groupId>
+			<artifactId>openmrs-api</artifactId>
+			<type>test-jar</type>
+		</dependency>
+
+		<dependency>
+			<groupId>org.openmrs.web</groupId>
```
Comment (dkayiwa) [inline api-pre2.2/pom.xml]: web in the api project?

**Comment by edrisse** [comment]
Hi @pluwum, what do I need to change to use using the version of EncounterDiagnosesElement of this module?

**Comment by dkayiwa** [comment]
@edrisse you do not need anything. It is automatically chosen depending on the version of the OpenMRS platform that you are running.

**Pull request merged**